phdi/fhir/conversion/convert.py

```
@@ -22,73 +23,77 @@
-            {
-                "name": "inputDataType",
-                "valueString": conversion_settings.get("input_data_type"),
-            },
-            {
-                "name": "templateCollectionReference",
-                "valueString": conversion_settings.get("template_collection"),
-            },
-            {
-                "name": "rootTemplate",
-                "valueString": conversion_settings.get("root_template"),
-            },
-        ],
+        "input_data": message,
+        "input_type": conversion_settings.get("input_type"),
+        "root_template": conversion_settings.get("root_template"),
     }
-    access_token = cred_manager.get_access_token().token
-    headers = {"Authorization": f"Bearer {access_token}"}
-
-    response = http_request_with_reauth(
-        cred_manager=cred_manager,
-        url=url,
-        retry_count=3,
-        request_type="POST",
-        allowed_methods=["POST"],
-        headers=headers,
-        data=data,
-    )
+
+    if cred_manager:
+        access_token = cred_manager.get_access_token().token
+        headers["Authorization"] = f"Bearer {access_token}"
+        response = http_request_with_reauth(
+            cred_manager=cred_manager,
+            url=url,
+            retry_count=3,
+            request_type="POST",
+            allowed_methods=["POST"],
+            headers=headers,
+            data=data,
+        )
+    else:
+        response = http_request_with_retry(
+            url=url,
+            retry_count=3,
+            request_type="POST",
+            allowed_methods=["POST"],
+            headers=headers,
+            data=data,
+        )
 
     if response.status_code != 200:
         raise Exception(
-            f"HTTP {str(response.status_code)} code encountered in $convert-data for a message"  # noqa
+            f"HTTP {str(response.status_code)} code encountered during conversion request"  # noqa
         )
     return response
 
```

Do we want to raise an exception here, or just return the response so that the user can see what went wrong during conversion? The converter provides relatively robust messaging back to users when they make bad requests.

---

I think it makes sense to return response regardless of status code. As an alternative, we could raise a requests.HTTPError that contains the response, so the caller can get to it if they need to.

IIRC @bryan-skylight might have been advocating for going the exception route, and @bamader might also have an opinion. What are your thoughts here?

---

I thought about this more and decided that wrapping in an HTTPError wasn't exactly correct, since the HTTP request actually succeeded, it is just reporting an error that occurred during the conversion operation. So I created a ConversionError, with an http_response property. Curious to know what you think of this approach!